docs/exts/taglist.py

```diff
@@ -0,0 +1,112 @@
+from sphinx.util import logging
+from docutils.parsers.rst import Directive
+from docutils import nodes
+from sphinx.environment import NoUri
+
+def initialize(app):
+    app.builder.env.tags = dict()
+
+class tag_node(nodes.General, nodes.Element): pass
+class taglist(nodes.General, nodes.Element):
+    tags = []
+
+class TagDirective(Directive):
+    '''
+    Directive for adding tags to a document
+    '''
+    has_content = False
+    required_arguments = 0
+    optional_arguments = 25
+
+    def run(self):
+        env = self.state.document.settings.env
+
+        tag = tag_node()
+        tag.tags = [tagname.strip().replace(',', '') for tagname in self.arguments]
+
+        return [tag]
+
+class TagListDirective(Directive):
+    '''
+    Directive for generating a list of tagged articles
```

Comment:
It might be obvious if one knows the Sphinx extensions domain, but this might deserve some more thorough comment, e.g. "Shinx calls this when ..., in order to... ".